Mapping array to exchange property fails for null

Issue #1:

1. Create mapping body-to-exchangeProperty (complex field) with source schema, that contains array field;
2. Send payload containing field with null value (instead of array, as described in source schema).

E.R.: Mapper skips fields with null values .
A.R.: Mapping process fails when attempting cast NullNode to ArrayNode.

`2025-05-08T16:37:06.372196505Z [2025-05-08T16:37:06.371] [ERROR] [request_id=1746722226272.0.1222803369768809] [tenant_id=00000000-0000-0000-0000-000000000000] [thread=0.0-8080-exec-6] [class=o.q.i.platform.engine.mapper.atlasmap.CustomAtlasContext    ] [traceId=-               ] [spanId=-               ] [originating_bi_id=-                                   ] [business_identifiers=-                                   ] [method=processSourceFieldMapping     ] [log_type=int] [session_id=e1717651-a9ea-4a64-9862-46403975122e] [chain_id=a028e0f6-5e34-40ef-8afb-a47186cae8e3] [chain=New Chain                     ] [chain_element_id=-                                   ] [chain_element=-              ] 
2025-05-08T16:37:06.372238765Z java.lang.ClassCastException: class com.fasterxml.jackson.databind.node.NullNode cannot be cast to class com.fasterxml.jackson.databind.node.ArrayNode (com.fasterxml.jackson.databind.node.NullNode and com.fasterxml.jackson.databind.node.ArrayNode are in unnamed module of loader org.springframework.boot.loader.launch.LaunchedClassLoader @3857f613)
2025-05-08T16:37:06.372241541Z 	at org.qubership.integration.platform.engine.mapper.atlasmap.json.QipAtlasJsonFieldReader.populateChildFields(QipAtlasJsonFieldReader.java:190)
2025-05-08T16:37:06.372243865Z 	at org.qubership.integration.platform.engine.mapper.atlasmap.json.QipAtlasJsonFieldReader.populateCollectionItems(QipAtlasJsonFieldReader.java:222)
2025-05-08T16:37:06.372245729Z 	at org.qubership.integration.platform.engine.mapper.atlasmap.json.QipAtlasJsonFieldReader.populateChildFields(QipAtlasJsonFieldReader.java:190)
2025-05-08T16:37:06.372247532Z 	at org.qubership.integration.platform.engine.mapper.atlasmap.json.QipAtlasJsonFieldReader.populateChildFields(QipAtlasJsonFieldReader.java:194)
2025-05-08T16:37:06.372249335Z 	at org.qubership.integration.platform.engine.mapper.atlasmap.json.QipAtlasJsonFieldReader.getJsonFieldsForPath(QipAtlasJsonFieldReader.java:101)`


Issue #2:

1. Create mapping object-to-object without detailed content schema;
2. Send a similar payload:
`{
    "body": {
        "phoneBook": {
            "contacts": [{
                "name": "John"
            }],
            "contacts": [{
                "name": "Mike"
            }],
            "contacts": null
        }
    }
}`

E.R.: `{
    "body": {
        "phoneBook": {
            "contacts": [{
                "name": "John"
            }],
            "contacts": [{
                "name": "Mike"
            }]
        }
    }
}`
A.R.: `{
    "body": {
        "phoneBook": {
            "contacts": null,
            "contacts": null
        }
    }
}`

--- a/src/main/java/org/qubership/integration/platform/engine/mapper/atlasmap/ComplexMappingAtlasModule.java
+++ b/src/main/java/org/qubership/integration/platform/engine/mapper/atlasmap/ComplexMappingAtlasModule.java
@@ -283,11 +283,11 @@ public abstract class ComplexMappingAtlasModule extends DelegatingAtlasModule {
             for (Field f : group.getField()) {
                 writeField(session, writer, f);
             }
-        } else {
+        } else if (nonNull(field.getValue())) {
             // If field is not found but has a value then we are updating the status
             // in order to get field processed by writer properly.
             // This is case of setting the default value.
-            if (nonNull(field.getValue()) && FieldStatus.NOT_FOUND.equals(field.getStatus())) {
+            if (FieldStatus.NOT_FOUND.equals(field.getStatus())) {
                field.setStatus(FieldStatus.SUPPORTED);
             }
             session.head().setTargetField(field);

--- a/src/main/java/org/qubership/integration/platform/engine/mapper/atlasmap/json/QipAtlasJsonFieldReader.java
+++ b/src/main/java/org/qubership/integration/platform/engine/mapper/atlasmap/json/QipAtlasJsonFieldReader.java
@@ -22,6 +22,7 @@ import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.MapperFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.NullNode;
 import io.atlasmap.api.AtlasConversionException;
 import io.atlasmap.api.AtlasException;
 import io.atlasmap.core.AtlasPath;
@@ -183,10 +184,10 @@ public class QipAtlasJsonFieldReader implements AtlasFieldReader {
         for (Field child : fieldGroup.getField()) {
             AtlasPath childPath = new AtlasPath(child.getPath());
             JsonNode childNode = node.get(childPath.getLastSegment().getName());
-            if (childNode == null) {
+            if (childNode == null || childNode instanceof NullNode) {
                 continue;
             }
-            if (childPath.getLastSegment().getCollectionType() != CollectionType.NONE && childNode instanceof ArrayNode) {
+            if (childPath.getLastSegment().getCollectionType() != CollectionType.NONE) {
                 FieldGroup childGroup = populateCollectionItems(session, (ArrayNode) childNode, child);
                 newChildren.add(childGroup);
             } else {

--- a/src/main/java/org/qubership/integration/platform/engine/mapper/atlasmap/json/QipAtlasJsonFieldReader.java
+++ b/src/main/java/org/qubership/integration/platform/engine/mapper/atlasmap/json/QipAtlasJsonFieldReader.java
@@ -186,7 +186,7 @@ public class QipAtlasJsonFieldReader implements AtlasFieldReader {
             if (childNode == null) {
                 continue;
             }
-            if (childPath.getLastSegment().getCollectionType() != CollectionType.NONE) {
+            if (childPath.getLastSegment().getCollectionType() != CollectionType.NONE && childNode instanceof ArrayNode) {
                 FieldGroup childGroup = populateCollectionItems(session, (ArrayNode) childNode, child);
                 newChildren.add(childGroup);
             } else {

